cloud_functions/climateiq_data_export_cf/main_test.py

```diff
@@ -46,11 +46,36 @@ def test_export_model_predictions_missing_study_area(
     }
     event = CloudEvent(attributes, data)
 
-    mock_study_area = mock.Mock()
-    mock_study_area.exists = False  # Indicate study area doesn't exist
+    # Build mock Storage object
+    mock_bucket = mock.create_autospec(storage.Bucket)
+    mock_blob = mock.create_autospec(storage.Blob)
+    predictions = '{"instance": [1], "prediction": [[1, 2, 3], [4, 5, 6]]}\n'
+    mock_fd = MagicMock()
+    mock_fd.__iter__.return_value = predictions.splitlines()
+    mock_blob.open.return_value.__enter__.return_value = mock_fd
+    mock_bucket.blob.return_value = mock_blob
+    mock_storage_client.return_value.bucket.return_value = mock_bucket
```

Comment:
The mocking library is actually quite a bit more flexible. You can replace all of these lines with
```python
    with mock_storage_client().bucket("").blob("").open() as mock_fd:
        mock_fd.__iter__.return_value = iter(predictions.splitlines())
```
there's no need to create & set each intermediate mock object for bucket & blob & so on. Same for the firestore mock below. I created a pull request demonstrating this:

https://github.com/UrbanSystemsLab/climateiq-frontend/pull/13/files

Reply:
oo nice, done!